chore: optimize prettier checks to only run on changed files

TylerBarnes opened the pull request:
Claude says:
This makes our prettier checks much faster by only checking files that have changed instead of the entire codebase.

The scripts are now composable, so you can run:
- `pnpm list-changed-files` to see all changed files
- `pnpm list-prettier-files` to filter for files prettier should format
- `pnpm prettier:changed` to format only changed files

The GitHub workflow now uses this faster approach too, which should speed up CI checks on PRs.

Also added a global `lint` command that was missing from the root package.json.

TylerBarnes committed:
perf: optimize prettier checks and add global lint command

- Add global lint command for running turbo lint across packages
- Add composable npm scripts for listing changed files
- Update GitHub workflow to check only changed files instead of entire repo
- Split prettier:changed into reusable list-changed-files and list-prettier-files scripts
- Add --log-level warn to prettier:format for cleaner output

wardpeet commented on package.json:
```diff
@@ -58,10 +58,14 @@
     "test:evals:llm": "pnpm --filter ./packages/evals test src/metrics/llm",
     "test:observability": "pnpm --filter ./observability/* test",
     "lint-staged": "lint-staged",
+    "lint": "turbo lint --filter \"./packages/**\" --filter \"./stores/**\" --filter \"./auth/**\" --filter \"./voice/**\" --filter \"./deployers/**\" --filter \"./workflows/**\" --filter \"./observability/**\" --filter \"./client-sdks/**\" --filter \"!@mastra/playground\"",
```
You probably want to do this instead
```suggestion
    "lint": "turbo --filter \"!./examples/**/*\" --filter \"!./docs/**/*\" --filter \"!./integrations/**/*\" --filter \"!@mastra/playground\" lint",
```

wardpeet commented on package.json:
```diff
@@ -58,10 +58,14 @@
-    "prettier:format": "prettier --write .",
+    "prettier:format": "prettier --write . --log-level warn",
     "prettier:format:check": "prettier --check .",
+    "list-changed-files": "(git diff --name-only --diff-filter=ACMR $(git symbolic-ref refs/remotes/origin/HEAD | sed 's@^refs/remotes/origin/@@')...HEAD && git diff --name-only --diff-filter=ACMR && git diff --cached --name-only --diff-filter=ACMR) | sort -u",
```
not sure if I trust this :D It takes about 44s in CI, not sure if it's worth the optimisation

TylerBarnes replied:
that's fair :) mostly useful locally where it takes half a second instead of 20

TylerBarnes committed:
fix: add fetch-depth:0 to checkout for git history access

As noted in PR review comment: https://github.com/mastra-ai/mastra/pull/6838#discussion_r1992470439
The workflow now requires full git history to run diff comparisons

TylerBarnes committed:
fix: update lint command to use turbo directly

As suggested in PR review comment: https://github.com/mastra-ai/mastra/pull/6838#discussion_r1992470440
Changed from 'pnpm turbo' to just 'turbo' for consistency

TylerBarnes committed:
fix: resolve CI failures for prettier checks

- Handle missing refs/remotes/origin/HEAD in CI by falling back to 'main'
- Use 'pnpm exec prettier' instead of bare 'prettier' in workflow
- Add error handling for git symbolic-ref command

Fixes CI failure: https://github.com/mastra-ai/mastra/actions/runs/17082977770

TylerBarnes committed:
revert: restore original GitHub workflow prettier check

Keep the local development optimizations but use full prettier check in CI
as suggested by wardpeet. The local commands still speed up development.
